Open-Rival/src/UnitAnimation.cpp

```diff
@@ -45,7 +50,14 @@ namespace Rival {
     }
 
     void UnitAnimation::tick() {
-        auto spritesheetSpan = spritesheetEntry.second - spritesheetEntry.first + 1;
-        animationStep = (animationStep + 1) % (spritesheetSpan);
+        auto spritesheetSpan = std::get<1>(spritesheetEntry) - std::get<0>(spritesheetEntry) + 1;
+        if (1 == spritesheetSpan) {
+            return;
+        }
+
+        while (getCurrentTime() > lastAnimationUpdate + std::get<2>(spritesheetEntry)) {
+            animationStep = (animationStep + 1) % (spritesheetSpan);
+            lastAnimationUpdate += std::get<2>(spritesheetEntry);
```

Comment:
I find all these get calls very confusing because they're accessing data without using any kind of human-readable name. Would it be possible to replace these with function calls or to store named fields in the animation class instead of relying purely on tuples?

Reply:
We could use structures which would clutter up our files. We could also have some wrappers for std::get<> methods but STL tuples are STL tuples... We can use [C++17 structured binding](https://en.cppreference.com/w/cpp/utility/tuple) too.

Reply:
I would argue that a structure that makes the code easier to read and maintain is not "clutter" :)

Reply:
Fair point :) So a structure, or would you tolerate std::tuple mechanisms?

Reply:
I would accept the code as-is, but I may get the itch to refactor it later to use a structure.

Reply:
I suggest to resist that urge because this is standard C++ :) I'll be okay with both ways though.